### PR TITLE
Add embed tests in CI

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Right now tests are running with --update-snapshots and thus they aren't verifying the snapshots but they are still useful in capturing bugs.

 
**Followup**
- Allow running the build manually and updating the snapshots and committing them to the main(using a PR)
- Run the tests without update-snapshots after that.

## Type of change

- Tests

### CODE REVIEW

**Comment by hariombalhara** [comment]
@zomars is there a way to make embed tests a non blocker for merge.

**Comment by zomars** [comment]
> @zomars is there a way to make embed tests a non blocker for merge.

You need to name it differently to `Testing 14.x and ubuntu-latest` Since that's a required check.

File: .github/workflows/e2e-embed.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
+name: E2E test - embed
+on:
+  push:
+    branches: [ tests/ci-embed ]
+  pull_request_target: # So we can test on forks
+    branches:
+      - main
+    paths-ignore:
+      - public/static/locales/**
```
Comment (zomars) [inline .github/workflows/e2e-embed.yml:9]: Also make sure to update these paths so this tests only run when embed-related code changes.

Comment (hariombalhara) [reply .github/workflows/e2e-embed.yml]: Thanks for the tip

File: .github/workflows/e2e-embed.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    branches:
+      - main
+    # Embed e2e - tests verify booking flow which is applicable to non-embed case also. So, don't ignore apps/web changes.
+    paths-ignore: 
+      - apps/api/**
+      - apps/console/**
+      - apps/docs/**
+      - apps/swagger/**
+      - apps/website/**
+      - public/**
```
Comment (zomars) [inline .github/workflows/e2e-embed.yml:15]: AFAIK we don't have a public folder on monorepo root? 🤔

Comment (hariombalhara) [reply .github/workflows/e2e-embed.yml:15]: True. It should have beens apps/web/public

Comment (hariombalhara) [reply .github/workflows/e2e-embed.yml]: Done

**Review by zomars — APPROVED** [review]
Feel free to merge if you feel this is ready @hariombalhara

**Comment by zomars** [comment]
Seems like the tests al running and failing @hariombalhara https://github.com/calcom/cal.com/runs/6659342595?check_suite_focus=true

**Comment by hariombalhara** [comment]
@zomars It is fixed in https://github.com/calcom/cal.com/pull/2929

**Pull request merged**